### PR TITLE
ci: release npm package on main branch

### PR DESCRIPTION
**Pull request by perfectmak**
This setups a github action to trigger releasing a package when merged
to master and the version in the package.json file changes.

### CODE REVIEW

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          cache: 'yarn'
+      - run: yarn install
+      - run: yarn build
+      - run: yarn test
+      - id: publish
+        name: Release to NPM
+        if: github.ref == 'refs/heads/master'
+        uses: JS-DevTools/npm-publish@v1
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
```
Comment (perfectmak) [inline .github/workflows/build-test-release.yml:20]: A valid NPM_TOKEN secret is required to be setup in this repository for this to work.

Comment (bonustrack) [reply .github/workflows/build-test-release.yml:20]: Would this publish on npm using beta version too?

Comment (perfectmak) [reply .github/workflows/build-test-release.yml:20]: It should use the version in the package.json file as the `latest` release version when publishing. Or do you mean something else?

Comment (perfectmak) [reply .github/workflows/build-test-release.yml:21]: I've included the beta tag now.

**Pull request merged**